File: src/TurkcellMeetingAssistant.Application/Common/Models/ApiResponse.cs
namespace TurkcellMeetingAssistant.Application.Common.Models;

public class ApiResponse<T>
{
    public T? Data { get; set; }
    public bool IsSuccess { get; set; }
    public string? Message { get; set; }
    public List<ApiError>? Errors { get; set; }

    public static ApiResponse<T> Success(T data, string? message = null)
    {
        return new ApiResponse<T>
        {
            Data = data,
            IsSuccess = true,
            Message = message
        };
    }

    public static ApiResponse<T> Fail(string message, List<ApiError>? errors = null)
    {
        return new ApiResponse<T>
        {
            IsSuccess = false,
            Message = message,
            Errors = errors
        };
    }
}

File: src/frontend/src/api/apiResponse.ts
export interface ApiResponse<T> {
  data: T | null;
  message: string | null;
  success: boolean;
  errors: string[] | null;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}
